lib/binding/kvdb_interface.c

```diff
@@ -445,14 +430,78 @@ hse_kvdb_close(struct hse_kvdb *handle)
     err = ikvdb_close((struct ikvdb *)handle);
     ev(err);
 
-    if (pidfile_remove(pfh) == -1 && !err)
-        err = merr(errno);
-
+    pidfile_remove(pfh);
     config_destroy(conf);
 
     return err;
 }
 
+hse_err_t
+hse_kvdb_attach(
+    const char *kvdb_home_tgt,
+    const char *kvdb_home_src,
+    const char *paths[HSE_MCLASS_COUNT])
+{
+    merr_t         err;
+    char           pidfile_path[PATH_MAX];
+    struct pidfh  *pfh_tgt = NULL, *pfh_src = NULL;
+    size_t         tlen, slen;
+
+    if (!kvdb_home_tgt || !kvdb_home_src) {
+        log_err("KVDB attach %s home must be provided", kvdb_home_src ? "target" : "source");
+        return merr(EINVAL);
+    }
+
+    tlen = strnlen(kvdb_home_tgt, PATH_MAX);
+    slen = strnlen(kvdb_home_src, PATH_MAX);
```

Comment:
If we check home paths for max len, we should probably also check mclass paths.

Reply:
Done.